android/java/org/chromium/chrome/browser/app/domain/TransactionsModel.java

```diff
@@ -0,0 +1,287 @@
+            BlockchainRegistry blockchainRegistry, JsonRpcService jsonRpcService,
+            EthTxManagerProxy ethTxManagerProxy, SolanaTxManagerProxy solanaTxManagerProxy,
+            BraveWalletService braveWalletService, AssetRatioService assetRatioService,
+            CryptoSharedData sharedData) {
+        mContext = context;
+        mTxService = txService;
+        mKeyringService = keyringService;
+        mBlockchainRegistry = blockchainRegistry;
+        mJsonRpcService = jsonRpcService;
+        mEthTxManagerProxy = ethTxManagerProxy;
+        mSolanaTxManagerProxy = solanaTxManagerProxy;
+        mBraveWalletService = braveWalletService;
+        mAssetRatioService = assetRatioService;
+        mSharedData = sharedData;
+        mAllNetworkInfoList = Collections.emptyList();
+        _mParsedTransactions = new MutableLiveData<>();
+        _mIsLoading = new MutableLiveData<>(false);
+        mIsLoading = _mIsLoading;
+        mParsedTransactions = _mParsedTransactions;
+        mAllAccountInfoList = Collections.emptyList();
+        addServiceObservers();
+    }
+
+    // Fetch all transactions of all accounts except rejected, and calculate balances, gas etc.
+    public void update(WeakReference<BraveWalletBaseActivity> activityRef) {
+        synchronized (mLock) {
+            _mIsLoading.postValue(true);
+            mActivityRef = activityRef;
+            if (JavaUtils.anyNull(
+                        mJsonRpcService, mKeyringService, mActivityRef, mActivityRef.get()))
+                return;
+            NetworkModel.getAllNetworks(mJsonRpcService, mSharedData.getSupportedCryptoCoins(), allNetworks -> {
+                mAllNetworkInfoList = new ArrayList<>(allNetworks);
+                KeyringModel.getKeyringInfos(mKeyringService, mSharedData.getEnabledKeyrings(), keyringInfos -> {
+                    mAllAccountInfoList = WalletUtils.getAccountInfosFromKeyrings(keyringInfos);
+                    var allAccountsArray = mAllAccountInfoList.toArray(new AccountInfo[0]);
+                    // Fetch transactions
+                    PendingTxHelper pendingTxHelper =
+                            new PendingTxHelper(mTxService, allAccountsArray, true, null);
+                    pendingTxHelper.fetchTransactions(() -> {
+                        HashMap<String, TransactionInfo[]> pendingTxInfos =
+                                pendingTxHelper.getTransactions();
+                        pendingTxHelper.destroy();
+                        TransactionInfo[] filteredTransactions =
+                                pendingTxInfos.values()
+                                        .stream()
+                                        .flatMap(
+                                                transactionInfos -> Arrays.stream(transactionInfos))
+                                        .filter(tx -> tx.txStatus != TransactionStatus.REJECTED)
+                                        .toArray(TransactionInfo[] ::new);
```

Comment:
Isn't Java 8 feature only available on Android 24+? Our lowest supported system version is Android 21 right? @SergeyZhukovsky

Reply:
I believe the minimum API is 24, https://bravesoftware.slack.com/archives/C03MMJ4CHEZ/p1675178531068709

Reply:
Just checked on play console, it seems the lowest level is 24+. Please ignore this comment.